### ISSUE TITLE
Fully deprecate the laplace inversion legacy flags

### ISSUE THREAD

**Issue by loeiten** [boutproject/BOUT-dev#235]
Two of the legacy flags are still in use, as seen around line `99` in `invert_laplace.hxx`:

``` cpp
const int INVERT_IN_RHS  = 16384; // Use input value in RHS at inner boundary
const int INVERT_OUT_RHS = 32768; // Use input value in RHS at outer boundary
```

They appear in  the `blob2d` examples (they also appear in `test-laplace2` and `test-petsc_laplace`).
As these are not mentioned in the `user_manual`, they can be confusing for users which wants to familiarize with laplace inversion using PETSc. 

I propose to remove these legacy flags (as has been done with the previous legacy flags), and use the new API flags in the mentioned examples (that is  using `outer_boundary_flags = 16` and `inner_ boundary_flags = 16` in the `BOUT.inp` files).

**Comment by d7919** [comment]
Can we update this issue/remove these legacy flags?

**Comment by ZedThree** [comment]
Now only in the `gyro-gem` example via calls to the gyro-average functions. We need a new interface for those functions to take the new flags really. Will aim for 4.3 so they can be removed in 5.0

**Comment by johnomotani** [comment]
Resolved by #1758.
